Add new bot: domain valid (#1966)
certtools/intelmq: docs/user/bots.rst

.. _intelmq.bots.experts.deduplicator.expert:

Domain valid
^^^^^^^^^^^^

**Information**

* `name:` `intelmq.bots.experts.domain_valid.expert`
* `lookup:` no
* `public:` yes
* `cache (redis db):` none

> * `description:` Valid first level domain from TLD list `tlds-alpha-by-domain.txt`. Latest list: https://data.iana.org/TLD/

Reviewer: That's not the only purpose, see the list of features that I collected below.
```suggestion
* `description:` Checks if a domain is valid by performing multiple validity checks (see below).
```

**Configuration Parameters**

>    * `domain_field`- event domain field

Reviewer: ```suggestion

- `domain_field`: The name of the field to be validated.
```
The empty line before the list is important, otherwise the list is not properly formatted.

>    * `tlds_domains_list` - local file location '/opt/intelmq/var/lib/bots/domain_valid/tlds-alpha-by-domain.txt'

Author: Oh, nice. I didn’t know about it.

**Description**

> Bot validating domain in event, if domain is not valid than event is dropped.

Reviewer: ```suggestion
If the field given in `domain_field` does not exist in the event, the event is dropped.
If the domain contains underscores (``_``), the event is dropped.
If the domain is not valid according to the `validators library <https://pypi.org/project/validators/>`_, the event is dropped.
If the domain's last part (the TLD) is not in the TLD-list configured by parameter ``tlds_domains_list``, the field is dropped.
```
Being more verbose what the bot actually does. This documentation is intended for users, so that they don't need to look at the code to figure out what a component is doing.

> Domain valid
> ^^^^^^^^^^^^
> 
> **Information**
> 
> * `name:` `intelmq.bots.experts.domain_valid.expert`
> * `lookup:` no
> * `public:` yes
> * `cache (redis db):` none
> * `description:` Valid first level domain from TLD list `tlds-alpha-by-domain.txt`. Latest list: https://data.iana.org/TLD/
> 
> **Configuration Parameters**
>    * `domain_field`- event domain field
>    * `tlds_domains_list` - local file location '/opt/intelmq/var/lib/bots/domain_valid/tlds-alpha-by-domain.txt'
> 
> **Description**
> Bot validating domain in event, if domain is not valid than event is dropped.
> 
> .. _intelmq.bots.experts.domain_valid.expert:
> 

Reviewer: ```suggestion

.. _intelmq.bots.experts.domain_valid.expert:

Domain valid
^^^^^^^^^^^^

**Information**

* `name:` `intelmq.bots.experts.domain_valid.expert`
* `lookup:` no
* `public:` yes
* `cache (redis db):` none
* `description:` Valid first level domain from TLD list `tlds-alpha-by-domain.txt`. Latest list: https://data.iana.org/TLD/

**Configuration Parameters**
   * `domain_field`- event domain field
   * `tlds_domains_list` - local file location '/opt/intelmq/var/lib/bots/domain_valid/tlds-alpha-by-domain.txt'

**Description**
Bot validating domain in event, if domain is not valid than event is dropped.

```
